R/summary_functions.R

```diff
@@ -133,6 +135,81 @@ zero_coverage <- function(x, ...) {
   }
 }
 
+
+
+
+#' Checks whether s1 contains s2
+#' s1: a srcref from branch_srcrefs
+#' s2: a srcref from covered_exp
+srcref_contains <- function (s1, s2) {
+    fst_ln1 <- s1[[1L]]
```

Comment:
I'm wondering isn't the following enough:
```
branch_start_line < expr_start_line || (branch_start_line == expr_start_line && branch_start_col <= expr_start_col)
```

Reply:
I think the main problem of the above code is assuming if `branch_start_line < expr_start_line` then the branch srcref contains the expression srcref. (true || anything == true) 

More specifically, replacing my code with the code above fails two of the tests I wrote for `srcref_contains`, which influences branch coverage results as well. The failing tests include calling another function in a `if` branch.  

I wrote two more tests, _test-now.R_ and _test-now2.R_, to compare the results. Both of them should report 50%, but with the code above 100%.

Reply:
I this the following is a counter example of your code.
```
f <- function (x) {
         x <- x + 1
         if (x > 0) {
              1
          } else {
               2
          }
       }
```
 Any branch srcref contains the srcref of `x <- x+1` , allowing branch coverage 100% when tested with f(0).